### ISSUE TITLE
Steam client waiting on worker thread stalled by a Proton 8.0 and MangoHUD interaction

### ISSUE THREAD

**Issue by Crozh** [ValveSoftware/Proton#6809]
#### Your system information

* Steam client version (build number or date): 1682708537 apr 28
* Distribution (e.g. Ubuntu): Arch 
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Steam client fails to shut down. This happens on an almost fresh install of Steam (What I mean by this is that I deleted the whole "root" folder and the Linux dependencies with all versions of Proton, except for "steamapps") explicitly after installing all the runtimes for Linux Gaming such as Proton, Sniper, Soldier and Steam Frameworks. 
Maybe it is correlated with the fact that Proton 8 is not working on my machine? I'm dumbfound at this point as to why this is a thing, but it's driving me crazy. Both problems have been a thing for a while on my system out of nowhere.  

#### Steps for reproducing this issue:

1. Install Steam using the official AUR repo
2. Check SteamPlay button
3. Install Steam dependencies such as the ones mentioned above. 
4. Try to install and play a game using proton 8?
5. Try to shut down Steam 
6. See task manager, type "Steam", now you have extra processes that weren't there before not wanting to shut down.
![2023-29-05-2](https://github.com/ValveSoftware/steam-for-linux/assets/44332979/386287d1-fa71-448b-9a69-d2ca59acd93e)

Update: Yes, just INSTALLING Proton 8, or even Proton GE 8-3/Experimental hangs Steam indefinitely. if I delete Proton 8 it gets fixed. This doesn't happen with Proton 7 installed.
What's going on with Proton 8? first version ran fine on my machine, but now it doesn't open any game and makes Steam not-shut down.

Another update: This is a mangohud related issue it seems. 
I checked the logs when I tried playing a game with Proton 8 and it showed a lot of config files from mangohud that were not found.  
I don't know which is to blame, but deleting mangohud fixes proton 8 which fixes the client issue.

**Comment by kisak-valve** [comment]
Hello @Crozh, from Steam's perspective, it's just waiting on a child process in a worker task to complete.

Since you've isolated this issue between Proton 8.0 and MangoHUD, I've transferred this issue report to the Proton issue tracker for a Proton dev to ponder. It may be worthwhile to also mention this to the MangoHUD dev(s).

**Comment by Crozh** [comment]
Yes, I figured later it was a child process and nothing more. But I find it strange that mangohud stops JUST Proton 8 from running/displaying games. I had Proton 7 installed and everything worked as usual. Maybe Proton 8 expects some files to be in a place that don't exist? from what I've seen on those logs. 
Maybe I'll check that problem later when I reinstall mangohud.

**Comment by rokam** [comment]
I've found that the issue was related to enabling MangoHud globally by adding `MANGOHUD=1`  to `/etc/environment`

**Comment by rokam** [comment]
https://github.com/flightlessmango/MangoHud/issues/1014
